sources/advanced/Cli.ts

```diff
@@ -304,7 +305,9 @@ export class Cli<Context extends BaseContext = BaseContext> implements MiniCli<C
                 ? commandClass.usage.examples.map(([label, cli]) => [formatMarkdownish(label, {format: this.format(colored), paragraphs: false}), cli.replace(/\$0/g, this.binaryName)])
                 : undefined;
 
-            data.push({path, usage, category, description, details, examples});
+            const options = builder.getOptions();
```

Comment:
I think that these options might be a bit too low level (`{names, description, arity, hidden, allowBinding}`). Wouldn't something like `{descriptor: '-v,--verbose', description: 'Show a more verbose greeting'}` make more sense? None of the other definitions are at the core level currently. 🤔 
I also think that the `hidden` properties should also be hidden from the definitions.

Reply:
> I also think that the hidden properties should also be hidden from the definitions.

Makes sense.

Not sure about renaming `names` to `definition` and joining it here? What do you think @arcanis ?

Reply:
I agree with @paul-soporan, let's just expose the basic information (using `formatMarksownish`), like the other fields 👍

I'm not sure I follow what you mean by renaming `names` - the way I see it, we'd just iterate over the options, and map them to the `{definition, description}` tuple.

Reply:
I was just wondering, why not keeping it with `names` as string array instead of doing a `names.join(',')` here. But I'll map it, no problem :).

Reply:
Do I need to use `formatMarkdownish` somewhere? Wasn't really sure 🤔 .

Reply:
> I was just wondering, why not keeping it with names as string array instead of doing a names.join(',') here. But I'll map it, no problem :).

My bad, by `definition` I didn't mean a simple join, but the complete definition: `-o,--output #0` (with the arity included). The `definition` is more useful than an array of names because it shows the actual usage of the option flag and is more similar to what the help page describes (I think that's the point of the definitions, to allow for documentation generation that looks to what's shown on the CLI help page, like we do in [the Yarn CLI docs](https://yarnpkg.com/cli/install)). For example, `{definition, description}` would allow us to generate a table of options and their descriptions inside Yarn's CLI docs.

> Do I need to use formatMarkdownish somewhere? Wasn't really sure 🤔 .

I think it might make sense to use it for the descriptions - it might be useful to highlight words between backticks inside descriptions. 🤔